### PR TITLE
[clang][bytecode] Implement comparsion operators for vector type

### PR DESCRIPTION
**Pull request by yronglin**
Implement ==, !=, <, <=, >, >= comparsion operators for vector type.

### CODE REVIEW

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (tbaederr) [inline clang/lib/AST/ByteCode/Compiler.cpp:1229]: Add an assertion that both LHS and RHS are also of vector type.

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (tbaederr) [inline clang/lib/AST/ByteCode/Compiler.cpp:1289]: I don't think we're going to use this anywhere else, so a separate function shouldn't be necessary

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (tbaederr) [inline clang/lib/AST/ByteCode/Compiler.cpp:1246]: Add an assertion if that's the case.

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (yronglin) [reply clang/lib/AST/ByteCode/Compiler.cpp:1229]: Done.

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (yronglin) [reply clang/lib/AST/ByteCode/Compiler.cpp:1248]: Done.

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (yronglin) [reply clang/lib/AST/ByteCode/Compiler.cpp]: Agree, removed.

File: clang/lib/AST/ByteCode/Compiler.cpp
Comment (tbaederr) [inline clang/lib/AST/ByteCode/Compiler.cpp:1267]: `ElemT` can be removed since it's always the same anyway

Comment (yronglin) [reply clang/lib/AST/ByteCode/Compiler.cpp]: Thanks, removed!

**Review by tbaederr — APPROVED** [review]

**Comment by yronglin** [comment]
Thanks for the review, wait for CI green before land.

**Pull request merged**
